core/src/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -205,6 +205,7 @@ protected void getPotentialLinks(Tile tile, Team team, Cons<Building> others){
             overlaps(tile.x * tilesize + offset, tile.y * tilesize + offset, other.tile(), laserRange * tilesize) && other.team == team &&
             !graphs.contains(other.power.graph) &&
             !PowerNode.insulated(tile, other.tile) &&
+            other.block.shouldConnect(other) &&
```

Comment:
I may not understand something , but why are you checking whether a block should connect to itself?